### PR TITLE
ADL-proof implementation of [alg.modifying.operations]

### LINKED ISSUE #4043: Calls to vector algorithms need qualification to avoid ADL
Per https://github.com/microsoft/STL/pull/3928/files#r1287824862, we need to qualify calls to the `__ugly` vector algorithms to avoid unintentional ADL e.g. when passing pointers-to-pointers-to-program-defined-types.

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Split from #4004, towards #140 and #1596.

Fixes #4043 by `::`-qualifying all remaining `__std_meow` vectorizing functions (some are `_STD`-qualified in #4146).

Also adds `_Is_trivially_ranges_swappable(_v)` to handle triviality of swapping by `ranges::swap` (which gets rid of ADL for pointers and pointers-to-members), along with workaround for DevCom-10456445.

### CODE REVIEW

File: tests/std/tests/P2538R1_adl_proof_std_projected/test.compile.pass.cpp [hunk 1]
```diff
@@ -210,6 +212,73 @@ void test_ranges_algorithms() {
...
+    validator varr2[1]{};
+
+    (void) copy_if(varr, varr + 1, varr2, simple_truth{});
+    (void) copy_if(varr, varr2, simple_truth{});
+    (void) copy_if(iarr, iarr + 1, iarr2, validating_truth{});
+    (void) copy_if(iarr, iarr2, validating_truth{});
+
+    (void) transform(varr, varr, varr2, std::identity{});
+    (void) transform(varr, varr2, std::identity{});
+    (void) transform(varr, varr, varr, varr2, varr2, simple_left_selector{});
```
Comment (StephanTLavavej) [inline tests/std/tests/P2538R1_adl_proof_std_projected/test.compile.pass.cpp:225]: These args appear to be wrong. This is a binary transform, so it's consuming the first range `[varr, varr)` (OK) and the second range `[varr, varr2)` (uh oh) and writing into `varr2`.

The other `transform` calls aren't wrong, but they're inconsistent. I think we should consistently consume `meow` as the one or two input ranges, and use `meow2` as the output range.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! I pushed a conflict-free merge with `main` plus a fix and consistency cleanups to `ranges::transform` calls in the tests - I'm only 99% sure about that change so please double-check.

File: tests/std/tests/P2538R1_adl_proof_std_projected/test.compile.pass.cpp (hunk 1 above)
Comment (frederick-vs-ja) [reply tests/std/tests/P2538R1_adl_proof_std_projected/test.compile.pass.cpp]: I believe your analysis and subsequent modifications are correct. Thanks for correction!

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
:shield: :tada: :cat2:
===

**Pull request merged**